Mimir/Controllers/TableSheetsController.cs

```diff
@@ -12,23 +9,38 @@ namespace Mimir.Controllers;
 public class TableSheetsController(TableSheetsRepository tableSheetsRepository) : ControllerBase
 {
     [HttpGet("names")]
-    public async Task<string[]> GetSheetNames(
-        string network
-    )
+    public async Task<string[]> GetSheetNames(string network)
     {
         var sheetNames = tableSheetsRepository.GetSheetNames(network);
 
         return sheetNames;
     }
 
     [HttpGet("{sheetName}")]
-    public async Task<ContentResult> GetSheet(
-        string network,
-        string sheetName
-    )
+    [Produces("application/json", "text/csv")]
+    public async Task<IActionResult> GetSheet(string network, string sheetName)
     {
-        var sheet = tableSheetsRepository.GetSheet(network, sheetName);
+        var acceptHeader = Request.Headers["Accept"].ToString();
 
-        return Content(sheet, "application/json");;
+        SheetFormat sheetFormat = acceptHeader.Contains("text/csv")
+            ? SheetFormat.Csv
+            : SheetFormat.Json;
+
+        try
+        {
+            var sheet = await tableSheetsRepository.GetSheet(network, sheetName, sheetFormat);
+
+            string contentType = sheetFormat switch
+            {
+                SheetFormat.Csv => "text/csv",
+                _ => "application/json"
+            };
```

Comment:
It seems better to implement extension method for `SheetFormat` enum type (e.g., `ToMimeType`) (Or handling with mime type instead of `SheetFormat`)

Reply:
i have created the SheetFormat extension method, thanks